### PR TITLE
[TEVA-2727] Fix subscription having email ending with .con

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-2727

### CODE REVIEW

File: lib/tasks/data.rake [hunk 1]
```diff
@@ -46,3 +46,34 @@ namespace :ons do
...
+      if Subscription.find_by(email: subscription.email.gsub(".con", ".com"))
+        subscription.destroy
+        deleted_subscriptions_count += 1
+      else
+        subscription.update_column(:email, subscription.email.gsub(".con", ".com"))
+        fixed_subscriptions_count += 1
+      end
+    end
+
+    subscriptions_to_delete.count
```
Comment (csutter) [inline lib/tasks/data.rake:71]: That should probably go somewhere?

File: lib/tasks/data.rake [hunk 2]
```diff
@@ -46,3 +46,34 @@ namespace :ons do
     %i[regions counties cities].each { |api_location_type| ImportPolygons.new(api_location_type: api_location_type).call }
   end
 end
+
+namespace :subscriptions do
+  desc "Fix subscriptions with wrong email"
+  task fix_wrong_email: :environment do
+    subscriptions_to_fix = Subscription.where("email LIKE '%.con' AND created_at > ?", 6.months.ago)
```
Comment (david-mears-dfe) [inline lib/tasks/data.rake:53]: I think Mili suggested 3 months, I'm inclined to agree

File: lib/tasks/data.rake [hunk 3]
```diff
@@ -46,3 +46,34 @@ namespace :ons do
...
+        subscription.update_column(:email, subscription.email.gsub(".con", ".com"))
+        fixed_subscriptions_count += 1
+      end
+    end
+
+    subscriptions_to_delete.count
+    subscriptions_to_delete.destroy_all
+
+    puts "Fixed #{fixed_subscriptions_count} subscriptions having email ending with .con"
+    puts "Deleted #{deleted_subscriptions_count} subscriptions having email ending with .con but also having the user created one with the right email"
```
Comment (josephhull676) [inline lib/tasks/data.rake:75]: Perhaps "Deleted #{deleted_subscriptions_count} subscriptions with an email ending with .con when the user has later created a subscription with the correct email address (ending in .com)" would sound better?

File: lib/tasks/data.rake [hunk 4]
```diff
@@ -46,3 +46,34 @@ namespace :ons do
...
+    puts "Also deleted #{subscriptions_to_delete.count} subscriptions having email ending with .con created more than 6 months ago"
```
Comment (josephhull676) [inline lib/tasks/data.rake:76]: "Also deleted #{subscriptions_to_delete.count} subscriptions created more than 6 months ago with an email ending with .con" sounds better to me

File: lib/tasks/data.rake [hunk 5]
```diff
@@ -46,3 +46,34 @@ namespace :ons do
...
+    puts "Have a nice day!"
```
Comment (josephhull676) [inline lib/tasks/data.rake:77]: Thanks, you too.

**Review by david-mears-dfe — APPROVED** [review]

**Pull request merged**